src/m365/entra/commands/group/group-user-add.ts

```diff
@@ -0,0 +1,192 @@
+import { Logger } from '../../../../cli/Logger.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { entraGroup } from '../../../../utils/entraGroup.js';
+import { entraUser } from '../../../../utils/entraUser.js';
+import { validation } from '../../../../utils/validation.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId?: string;
+  groupDisplayName?: string;
+  ids?: string;
+  userNames?: string;
+  role: string;
+}
+
+class EntraGroupUserAddCommand extends GraphCommand {
+  private readonly roleValues = ['Owner', 'Member'];
+
+  public get name(): string {
+    return commands.GROUP_USER_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a user to a Microsoft Entra ID group';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupDisplayName: typeof args.options.groupDisplayName !== 'undefined',
+        ids: typeof args.options.ids !== 'undefined',
+        userNames: typeof args.options.userNames !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --groupId [groupId]'
+      },
+      {
+        option: '-n, --groupDisplayName [groupDisplayName]'
+      },
+      {
+        option: '--ids [ids]'
+      },
+      {
+        option: '--userNames [userNames]'
+      },
+      {
+        option: '-r, --role <role>',
+        autocomplete: this.roleValues
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID for option groupId.`;
+        }
+
+        if (args.options.ids) {
+          const ids = args.options.ids.split(',').map(i => i.trim());
+          if (!validation.isValidGuidArray(ids)) {
```

Comment:
Let's update this in a separate issue so that we don't need to iterate through the array twice